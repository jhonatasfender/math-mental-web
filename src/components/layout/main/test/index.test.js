import Main from '@components/layout/main';
import { screen } from '@testing-library/dom';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';

describe('<Main />', () => {
  it('should render', () => {
    const { container } = render(
      <ThemeProvider theme={{ sector: 'addition' }}>
        <Main />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should click the button and switch to dark theme', () => {
    const { container } = render(
      <ThemeProvider theme={{ sector: 'addition' }}>
        <Main />
      </ThemeProvider>,
    );

    const buttonDark = screen.getByRole('button', { name: 'dark mode' });
    expect(buttonDark).toBeInTheDocument();
    userEvent.click(buttonDark);

    /**
     * validar se o tema realmente foi mudado
     */
    expect(container).toMatchSnapshot();
  });
});
