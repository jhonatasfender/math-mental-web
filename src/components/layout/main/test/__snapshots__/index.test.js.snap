// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should click the button and switch to dark theme 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 3rem;
}

.c3 {
  padding: 1rem 17rem;
  width: 100wh;
  background: #7f7f7f;
}

.c3 * {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

<div>
  <header
    class="c0"
  >
    <svg
      aria-hidden="false"
      aria-label="dark mode"
      class="c1 c2"
      fill="currentColor"
      focusable="false"
      role="button"
      type="button"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a10 10 0 1 0 0-20 10 10 0 0 0 0 20zm0-2V4a8 8 0 1 1 0 16z"
      />
    </svg>
  </header>
  <div
    class="c3"
  />
</div>
`;

exports[`<Main /> should render 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 3rem;
}

.c3 {
  padding: 1rem 17rem;
  width: 100wh;
  background: #EDEDED;
}

.c3 * {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

<div>
  <header
    class="c0"
  >
    <svg
      aria-hidden="false"
      aria-label="dark mode"
      class="c1 c2"
      fill="currentColor"
      focusable="false"
      role="button"
      type="button"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 22a10 10 0 1 0 0-20 10 10 0 0 0 0 20zm0-2V4a8 8 0 1 1 0 16z"
      />
    </svg>
  </header>
  <div
    class="c3"
  />
</div>
`;
